sleap/nn/system.py

```diff
@@ -183,3 +184,46 @@ def best_logical_device_name() -> Text:
         cpus = tf.config.list_logical_devices("CPU")
         device_name = cpus[0].name
     return device_name
+
+
+def get_gpu_memory() -> dict:
+    """Return dictionary of GPU(s) and current vRAM usage on a machine.
+
+    Returns:
+        Dictionary of GPU indexes (key) and percent of memory available (values)
+
+    Notes:
+        A value of 1.0 means all memory is available, or 100%.
+    """
+    command = [
+        "nvidia-smi",
+        "--query-gpu=index,memory.free,memory.total",
+        "--format=csv",
+    ]
+
+    memory_poll = subprocess.run(command, capture_output=True)
+
+    # Capture subprocess standard output
+    subprocess_result = memory_poll.stdout
+
+    # nvidia-smi returns an ascii encoded byte string separated by newlines (\n)
+    # Splitting gives a list where the final entry is an empty string. Slice it off
+    # and finally slice off the csv header in the 0th element.
+    memory_string = subprocess_result.decode("ascii").split("\n")[:-1][1:]
+
+    memory_dict = {}
+
+    for row in memory_string:
+        # Get the gpu ID and the available/total memory
+        gpu_id = row.split(",")[0]
+        available_memory = row.split(",")[1]
+        total_memory = row.split(",")[2]
+
+        # Removing megabyte text returned by nvidia-smi
+        available_memory = available_memory.split()[0]
+        total_memory = total_memory.split()[0]
+
+        # Append percent of GPU available to GPU ID
+        memory_dict[gpu_id] = round(int(available_memory) / int(total_memory), 4)
```

Comment:
Do we want this to be a percentage?

Reply:
I don't know! I wasn't sure if having it as a percentage would be helpful/more readable/understandable so I just left it that way.

Reply:
I also realize now that the comment says percentage but it's not a percent value that's given, so we can change the comment or the value. I have no preference and also don't know what best practice is lol

Reply:
I think as it is currently written it is a fraction no? Ratio of:

Available Memory
________________
Total Memory